### PR TITLE
PeerDAS: Re-enable full node joining the main fork

### PR DESCRIPTION
**Pull request by nalepae**
**Please read commit by commit. Important commits are the two last ones.**

Before this pull request, if, for example, a data column was not received by a full node peer, then this full node peer was unable to re-join the main fork.

Why?
Because, in order to consider a block as available, the full node peer needs the 4 custody columns + the 4 extra subnet sampling columns.
However, before this PR, when requesting data columns by root, the node was requesting **only** the 4 custody columns.

This PR fixes that.

In the following example, `4-geth-prysm` ignores all incoming data columns for slots 10 and 20.
Depending how close the next block proposal is (if the node had time to rejoin), then we see a fork.

However, each time, it rejoins the main fork (slots 15 and 25 are not orphaned.)

image

### CODE REVIEW

File: beacon-chain/sync/subscriber.go [hunk 1]
```diff
@@ -334,7 +332,6 @@ func (s *Service) wrapAndReportValidation(topic string, v wrappedVal) (string, p
 					"topic":        topic,
 					"multiaddress": multiAddr(pid, s.cfg.p2p.Peers()),
 					"peerID":       pid.String(),
-					"agent":        agentString(pid, s.cfg.p2p.Host()),
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go:337]: why is this removed ? its very useful on mainnet to catch misbehaving clients

Comment (nalepae) [reply beacon-chain/sync/subscriber.go]: The reason of removal is written in the corresponding [commit message](https://github.com/prysmaticlabs/prysm/pull/14475/commits/55db3511022b14e43ad65d8c61240db4ae56f034).

I removed it because it was quite useless on devnet, since we know who is who.
But yes, on mainnet/testnet, it makes sense.

Fixed in 0cf7d4fcdd5560611ba48ece95903fdeefaded02.
